test/results/tiktok.py

```diff
@@ -0,0 +1,183 @@
+    "#comment"  : "/video/ link: few photos",
+    "#category" : ("", "tiktok", "post"),
+    "#class"    : tiktok.TiktokPostExtractor,
+    "#pattern"  : PATTERN,
+    "#options"  : {"videos": False}
+},
+{
+    "#url"      : "https://vm.tiktok.com/ZGdhVW3cu/",
+    "#comment"  : "vm.tiktok.com link: few photos",
+    "#category" : ("", "tiktok", "vmpost"),
+    "#class"    : tiktok.TiktokVmpostExtractor,
+    "#pattern"  : PATTERN,
+    "#options"  : {"videos": False}
+},
+{
+    "#url"       : "https://www.tiktok.com/@ughuwhguweghw/video/1",
+    "#comment"   : "deleted post",
+    "#category"  : ("", "tiktok", "post"),
+    "#class"     : tiktok.TiktokPostExtractor,
+    "#exception" : exception.NotFoundError,
+    "#options"  : {"videos": False}
+},
+{
+    "#url"      : "https://www.tiktok.com/@memezar/video/7449708266168274208",
+    "#comment"  : "Video post",
+    "#category" : ("", "tiktok", "post"),
+    "#class"    : tiktok.TiktokPostExtractor,
+    "#pattern"  : PATTERN_WITH_AUDIO,
+    "#options"  : {"videos": True}
+},
+{
+    "#url"      : "https://www.tiktok.com/@memezar/photo/7449708266168274208",
+    "#comment"  : "Video post as a /photo/ link",
+    "#category" : ("", "tiktok", "post"),
+    "#class"    : tiktok.TiktokPostExtractor,
+    "#pattern"  : PATTERN_WITH_AUDIO,
+    "#options"  : {"videos": True}
+},
+{
+    "#url"      : "https://vm.tiktok.com/ZGdht7cjp/",
+    "#comment"  : "Video post as a VM link",
+    "#category" : ("", "tiktok", "vmpost"),
+    "#class"    : tiktok.TiktokVmpostExtractor,
+    "#pattern"  : PATTERN_WITH_AUDIO,
+    "#options"  : {"videos": True}
+},
+{
+    "#url"      : "https://www.tiktok.com/@memezar/video/7449708266168274208",
+    "#comment"  : "Skipping video post",
+    "#category" : ("", "tiktok", "post"),
+    "#class"    : tiktok.TiktokPostExtractor,
+    "#pattern"  : PATTERN,
+    "#options"  : {"videos": False}
+},
+{
+    "#url"      : "https://www.tiktok.com/@chillezy/photo/7240568259186019630",
+    "#comment"  : "/photo/ link: many photos with audio",
+    "#category" : ("", "tiktok", "post"),
+    "#class"    : tiktok.TiktokPostExtractor,
+    "#pattern"  : PATTERN_WITH_AUDIO,
+    "#options"  : {"videos": True}
+},
+{
+    "#url"      : "https://www.tiktok.com/@chillezy/video/7240568259186019630",
+    "#comment"  : "/video/ link: many photos with audio",
+    "#category" : ("", "tiktok", "post"),
+    "#class"    : tiktok.TiktokPostExtractor,
+    "#pattern"  : PATTERN_WITH_AUDIO,
+    "#options"  : {"videos": True}
+},
+{
+    "#url"      : "https://vm.tiktok.com/ZGdh4WUhr/",
+    "#comment"  : "vm.tiktok.com link: many photos with audio",
+    "#category" : ("", "tiktok", "vmpost"),
+    "#class"    : tiktok.TiktokVmpostExtractor,
+    "#pattern"  : PATTERN_WITH_AUDIO,
+    "#options"  : {"videos": True}
+},
+{
+    "#url"      : "https://www.tiktok.com/@chillezy",
+    "#comment"  : "User profile",
```

Comment:
I should admit that user profile scraping can be a tiny bit flakey. Very occasionally `yt-dlp` will fail to extract user profile information and there's not a lot `gallery-dl` can do about it. Is it okay to keep these tests in, or should they be skipped?